src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -450,6 +452,7 @@ export class CommandDetectionCapability implements ICommandDetectionCapability {
 			const endMarker = this._currentCommand.commandFinishedMarker;
 			const newCommand: ITerminalCommand = {
 				command: this._handleCommandStartOptions?.ignoreCommandLine ? '' : (command || ''),
+				rawCommandLine: this._handleCommandStartOptions?.ignoreCommandLine ? '' : (this._currentCommand.rawCommandLine || ''),
```

Comment:
As discussed this approach goes against `633 E` which is meant to be the source of truth. We should try send the right command before aliases are resolved if possible, if not maybe we need to think about alternatives